xrpl/models/transactions/account_delete.py

```diff
@@ -42,3 +44,22 @@ class AccountDelete(Transaction):
         default=TransactionType.ACCOUNT_DELETE,
         init=False,
     )
+
+    credential_ids: Optional[List[str]] = None
+    """Credentials associated with sender of this transaction. The credentials included
+    must not be expired. The list must not be empty when specified and cannot contain
+    more than 8 credentials."""
+
+    def _get_errors(self: Self) -> Dict[str, str]:
+        errors = super()._get_errors()
+
+        # Validation checks on the credential_ids field
+        if self.credential_ids is not None:
+            if len(self.credential_ids) == 0:
+                errors["credential_ids"] = "CredentialIDs list cannot be empty."
+            if len(self.credential_ids) > 8:
+                errors[
+                    "credential_ids"
+                ] = "CredentialIDs list cannot have more than 8 elements."
```

Comment:
I would suggest putting this in a helper function so that you don't need to duplicate the code across each of the transactions. It could even potentially be added to the parent `_get_errors` function if you get a bit clever about how you check for the field.

Reply:
I tried to throw this error in the parent function, I got this Python interpreter error: `Instance of 'Transaction' has no 'credential_ids' member`

Reply:
Yeah, that's what I meant by getting clever - e.g. you could use `self.__dict__` or something instead.

Reply:
refactored in bf5114a5

Reply:
@mvadari  Hmm okay, I didn't think of `self.__dict__`. From a readability point-of-view, I like the current refactored version. I don't see a need to push the error-validation up in `Transaction` class.